tests/core/test_stellargraph.py

```diff
@@ -427,3 +428,71 @@ def test_feature_conversion_from_iterator():
     ab = gs.get_feature_for_nodes([4, 5], "B")
     assert ab.shape == (2, 10)
     assert ab[:, 0] == pytest.approx([4, 5])
+
+
+def example_benchmark_graph(feature_size=None, n_nodes=100, n_edges=200, n_types=4):
+    G = nx.Graph()
+
+    G.add_nodes_from(range(n_nodes))
+    edges = [
+        (random.randint(0, n_nodes - 1), random.randint(0, n_nodes - 1))
+        for _ in range(n_edges)
+    ]
+    G.add_edges_from(edges)
+
+    # Add example features
+    if feature_size is not None:
+        for v in G.nodes():
+            G.nodes[v]["feature"] = np.ones(feature_size)
+            G.nodes[v]["label"] = v % n_types
+
+        G = StellarGraph(G, node_features="feature")
+    else:
+        G = StellarGraph(G)
+    return G
+
+
+@pytest.mark.benchmark(group="StellarGraph neighbours")
+def test_benchmark_get_neighbours(benchmark):
+    g = example_benchmark_graph()
+    num_nodes = g.number_of_nodes()
+
+    # get the neigbours of every node in the graph
+    def f():
+        for i in range(num_nodes):
+            g.neighbors(i)
+
+    benchmark(f)
+
+
+@pytest.mark.benchmark(group="StellarGraph node features")
+@pytest.mark.parametrize("num_types", [1, 4])
+@pytest.mark.parametrize("lookup", ["individual", "bulk"])
+@pytest.mark.parametrize("type_arg", ["infer", "specify"])
```

Comment:
This gives a large number of test combinations. My suggestion is to drop the "lookup" option and have randomly sampled batches of a fixed number of nodes of either any type (with "infer") or the same type (with "specify")

Reply:
I've updated the test along those lines (and updated the PR description with the new output).